### PR TITLE
Create track model

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
This adds a model which provides track data using `kwiver::track` as the underlying data structure.

### CODE REVIEW

File: sealtk/core/KwiverTrackModel.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  }
+}
+
+// ----------------------------------------------------------------------------
+void KwiverTrackModel::setTracks(
+  kv::object_track_set_sptr const& trackSet)
+{
+  this->beginResetModel();
+
+  with_expr(qtScopedBlockSignals{this})
```
Comment (KyleFromKitware) [inline sealtk/core/KwiverTrackModel.cpp]: `with_expr` causes build errors on my machine.

Comment (mwoehlke-kitware) [reply sealtk/core/KwiverTrackModel.cpp]: Oh? What errors?

ViViA has been using it for some time...

File: sealtk/core/test/KwiverTrackModel.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        createTrack(4, track4),
+        createTrack(5, track5),
+      }));
+  QCOMPARE(model.rowCount(), 5);
+
+  testTrackData(model, 0, 1, track1);
+  testTrackData(model, 1, 2, track2);
+  testTrackData(model, 2, 3, track3);
+  testTrackData(model, 3, 4, track4);
+  testTrackData(model, 4, 5, track5);
```
Comment (KyleFromKitware) [inline sealtk/core/test/KwiverTrackModel.cpp]: Is there a reason this test can't be split into "data" and "test" parts the way other Qt tests are?

Comment (mwoehlke-kitware) [reply sealtk/core/test/KwiverTrackModel.cpp]: What would be the effect of that, except possibly to move the definitions of `trackN` from an anonymous namespace to `operations_data()`? Is that worth the extra "frills"?

(Note: it is absolutely intended that the test tests all five tracks at once... the objective is to test a model with *multuple* tracks, not just to individually test different tracks.)

Personally I am inclined to only use the test/data split if I am going to run the same test code on more than one set of data. For that use case, it is definitely valuable. For a single data set, however, I'm not convinced it is worthwhile. (OTOH, if you have suggestions how this test would be improved by using multiple data sets...?)

**Pull request merged**